src/tests/classic/run.rs

```diff
@@ -281,6 +283,46 @@ fn test_forms_of_destructuring_allowed_by_classic_1() {
     );
 }
 
+fn run_dependencies(filename: &str) -> HashSet<String> {
+    let result_text = do_basic_run(&vec![
+        "run".to_string(),
+        "-i".to_string(),
+        "resources/tests".to_string(),
+        "-M".to_string(),
+        filename.to_owned(),
+    ])
+    .trim()
+    .to_string();
+
+    eprintln!("run_dependencies:\n{}", result_text);
+
+    let mut dep_set = HashSet::new();
+    for l in result_text.lines() {
+        if let Some(suffix_start) = l.find("resources/tests") {
+            let copied_suffix: Vec<u8> = l.as_bytes().iter().skip(suffix_start).copied().collect();
+            dep_set.insert(decode_string(&copied_suffix));
+        } else {
+            panic!("file {} isn't expected", l);
+        }
+    }
+
+    dep_set
+}
+
+#[test]
+fn test_get_dependencies_1() {
+    let dep_set = run_dependencies("resources/tests/singleton_top_layer.clvm");
+
+    eprintln!("dep_set {dep_set:?}");
+
+    let mut expect_set = HashSet::new();
+    expect_set.insert("resources/tests/condition_codes.clvm".to_owned());
+    expect_set.insert("resources/tests/curry-and-treehash.clinc".to_owned());
+    expect_set.insert("resources/tests/singleton_truths.clib".to_owned());
```

Comment:
I read this as `tests/singleton_top_layer.clvm` refers to this file, "singleton_truths.clib". But this file was just added in this PR. Prior to this change, was `singleton_top_layer.clvm` not exercised, since it was missing a dependency?

Reply:
It was used in the use checker, and usecheck-work and usecheck-fail contained the real version, and a version with a bug of curry-and-treehash.clinc.  It was previously here to verify that the use checker can look at a complex real world program and see that an unused value in a leaf function lead to detection of an unused parameter in the whole module.  That feature was originally conceived by the chialisp code audit people.